### PR TITLE
Standardized Code Climate Git Action

### PR DESCRIPTION
**Pull request by akroon3r**
Created a standard git action for the BCParis Service repository which should work here.

No more (pambaati) dependencies.

### CODE REVIEW

**Comment by akroon3r** [comment]
whoops need the test reporter ID

File: .github/workflows/code-climate-test-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      
+      # Formatting the BACKEND coverage reports generated (dynamically)
+      - name: Format BACKEND coverage reports
+        run: |
+          JACOCO_SOURCE_PATH=${{ github.workspace }}/src/main/java ./cc-test-reporter format-coverage ${{github.workspace}}/target/site/jacoco/jacoco.xml --input-type jacoco --output coverage/bcparis-codeclimate.json;
+
+      # Upload total coverage report to Code Climate
+      - name: Upload coverage report to Code Climate
+        run: ./cc-test-reporter upload-coverage -i coverage/bcparis-codeclimate.json
+
```
Comment (sdevalapurkar-bcgov) [inline .github/workflows/code-climate-test-coverage.yml:86]: github seems to saying no newline even tho i see one

**Review by sdevalapurkar-bcgov — DISMISSED** [review]

File: .github/workflows/code-climate-test-coverage.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
```
Comment (akroon3r) [inline .github/workflows/code-climate-test-coverage.yml:86]:
```suggestion
```

File: .github/workflows/code-climate-test-coverage.yml (hunk 1 above)
Comment (akroon3r) [reply .github/workflows/code-climate-test-coverage.yml]: weird. should be resolved now?

File: .github/workflows/code-climate-test-coverage.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        run: mvn -B verify -P all --file pom.xml
+      
+      # Formatting the BACKEND coverage reports generated (dynamically)
+      - name: Format BACKEND coverage reports
+        run: |
+          JACOCO_SOURCE_PATH=${{ github.workspace }}/src/main/java ./cc-test-reporter format-coverage ${{github.workspace}}/target/site/jacoco/jacoco.xml --input-type jacoco --output coverage/bcparis-codeclimate.json;
+
+      # Upload total coverage report to Code Climate
+      - name: Upload coverage report to Code Climate
+        run: ./cc-test-reporter upload-coverage -i coverage/bcparis-codeclimate.json
```
Comment (sdevalapurkar-bcgov) [inline .github/workflows/code-climate-test-coverage.yml:85]:
```suggestion
        run: ./cc-test-reporter upload-coverage -i coverage/splunk-chatops-codeclimate.json
```

File: .github/workflows/code-climate-test-coverage.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+          ls
+        
+      # Run Maven Verify to generate all jacoco reports
+      - name: Build with Maven
+        run: mvn -B verify -P all --file pom.xml
+      
+      # Formatting the BACKEND coverage reports generated (dynamically)
+      - name: Format BACKEND coverage reports
+        run: |
+          JACOCO_SOURCE_PATH=${{ github.workspace }}/src/main/java ./cc-test-reporter format-coverage ${{github.workspace}}/target/site/jacoco/jacoco.xml --input-type jacoco --output coverage/bcparis-codeclimate.json;
```
Comment (sdevalapurkar-bcgov) [inline .github/workflows/code-climate-test-coverage.yml:81]:
```suggestion
          JACOCO_SOURCE_PATH=${{ github.workspace }}/src/main/java ./cc-test-reporter format-coverage ${{github.workspace}}/target/site/jacoco/jacoco.xml --input-type jacoco --output coverage/splunk-chatops-codeclimate.json;
```

**Review by sdevalapurkar-bcgov — CHANGES_REQUESTED** [review]

**Review by adodge-bcg — DISMISSED** [review]

File: .github/workflows/code-climate-test-coverage.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
+name: Generate/Upload all Coverage Reports
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```
Comment (adodge-bcg) [inline .github/workflows/code-climate-test-coverage.yml:7]:
```suggestion
    branches: [main]
  pull_request:
    branches: [main]
```

**Review by adodge-bcg — APPROVED** [review]

**Review by sdevalapurkar-bcgov — APPROVED** [review]
thanks, sorry for delay in approval

**Pull request merged**